File: EOB cup cake task/Cakes.java
public class Cakes{
	public static int findSmallest(int array[]){
	
	

int smallest = array[0];

for(int count = 0; count <= (array.length - 1); count++){
	
	if (array[count] < smallest){
	smallest = array[count];
	}
}

return smallest;


} 

public static int average(int array[]){
	

int average = 0;
int sum = 0;
for(int count = 0; count < array.length ; count++){

	sum += array[count];
	average = sum / array.length;
	
}

return average;

}


public static int occurence(int array[], int number){
 
int occurence = 0;	
for(int count = 0; count < array.length ; count++){
	if(number == array[count])
	 occurence += 1;
	 
	
}

return occurence;


}

public static boolean containsElement(int array[], int number){


for(int count = 0; count < array.length; count++){
	
	if (array[count] == number){
	
	return true;
	}
	}

	return false;
}

public static int firstElement(int array[]){
	

for(int count = 0; count <= (array.length); count++){
	
	return array[0];
	
}

return 0;
}

public static int lastElement(int array[]){
	

for(int count = 0; count <= (array.length); count++){

	 return array[8];
		
}

 return 0;
}



public static int arrayLength(int array[]){
	

for(int count = 0; count <= (array.length); count++){

		
}
	 return array[4];	
}
}

File: EOB Cup cake Python/cakes.py
def find_smallest(array):
    smallest = array[0]

    for count in range(len(array)):
        if array[count] < smallest:
            smallest = array[count]
    
    return smallest



def average(array):
    total = 0
    for count in range(len(array)):
        total += array[count]
    average = total / len(array)
    return average



def occurrence(array, number):
    count_occurrence = 0
    for count in range(len(array)):
        if array[count] == number:
            count_occurrence += 1
    return count_occurrence



def contains_element(array, number):
    for count in range(len(array)):
        if array[count] == number:
            return True
    return False








numbers = [10, 20, 30, 40, 50, 20, 10, 80, 90]

print("Smallest:", find_smallest(numbers))
print("Average:", average(numbers))
print("Occurrence of 20:", occurrence(numbers, 20))
print("Contains 100?", contains_element(numbers, 100))
